pkg/cache/cache.go

```diff
@@ -220,18 +220,35 @@ func (c *Cache) WaitForPodsReady(ctx context.Context) {
 	defer c.Unlock()
 
 	log := ctrl.LoggerFrom(ctx)
+
+	podsReadyCond := func(ch chan any) {
+		c.Lock()
+		defer c.Unlock()
+		defer close(ch)
+		c.podsReadyCond.Wait()
+		ch <- struct{}{}
+	}
+
+	podsReadyCondCh := make(chan any)
```

Comment:
```suggestion
	podsReadyCondCh := make(chan struct{})
	defer close(podsReadyCondCh)
```

Can we use `struct{}` typed to reduce the amount of memory?
Also, should we close this channel?

Reply:
sure, changed to struct type. Yeah, the channel is already closed [inside goroutine](https://github.com/kubernetes-sigs/kueue/pull/915/files#diff-964e351ee2375d359c78d69e514c4edc42577219761c4475f391ed2daf715e51R128)

Reply:
> the channel is already closed inside goroutine

Ah, I see. I missed the line. Also, we shouldn't call `defer` inside the infinity loop since the memory is caused by the `defer`.

So, your code looks good, thanks!